src/main/java/io/reactivex/rxjava3/exceptions/MissingBackpressureException.java

```diff
@@ -20,6 +20,25 @@ public final class MissingBackpressureException extends RuntimeException {
 
     private static final long serialVersionUID = 8517344746016032542L;
 
+    /**
+     * The default error message.
+     * <p>
+     * This can happen if the downstream doesn't call {@link org.reactivestreams.Subscription#request(long)}
+     * in time or at all.
+     * @since 3.1.6
+     */
+    public static final String DEFAULT_MESSAGE = "Could not emit value due to lack of requests";
+
+    /**
+     * The message for queue overflows.
+     * <p>
+     * This can happen if the upstream disregards backpressure completely or calls
+     * {@link org.reactivestreams.Subscriber#onNext(Object)} concurrently from multiple threads
+     * without synchronization. Rarely, it is an indication of bugs inside RxJava
+     * @since 3.1.6
+     */
+    public static final String QUEUE_OVERFLOW_MESSAGE = "Queue overflow due to illegal concurrent onNext calls or a bug in RxJava";
```

Comment:
These could be private though, right?

Reply:
3rd party implementations should be able to verify they got this exact message:

```java
assertEquals(ex.getMessage(), MissingBackpressureException.QUEUE_OVERFLOW_MESSAGE);
```

Reply:
Hmm should not we then just have two different exceptions both having only one public constructor which contains the exact message and then you simply check for its type which is superior than the String message?

Reply:
Alright, let's introduce a separate exception.